### PR TITLE
[REVIEW] Moving cuML prims to RAFT

### PR DESCRIPTION
**Pull request by divyegala**
This PR moves cuml prims to RAFT that will break the circular dependency between libcuml and libcumlprims. The changes made in these files are *almost* exactly the same as those mirrored by cuML, with only the header paths updated to reflect how raft does it:

1. Like `#include <common/cudart_utils.h>` became `#include <raft/cudart_utils.h>`
2. "#include "test_utils.h" became "../test_utils.h" (Our prims gtests are all in the same, root level `tests/prims` folder but I took the liberty of dividing it more as `tests/linalg`, `tests/stats`, etc.)

In any case, these changes will not affect when we finally delete prims from cuML. Only thing we would need to do there is also update the include path for these prims (files moved), as they are already in the correct formatting and namespaces.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by JohnZed** [comment]
We discussed offline that the build changes are new, but can you clarify the other files - are they all _exact_ copies of the recently-updated cuml versions? Or are there additional changes?

**Comment by divyegala** [comment]
@JohnZed updated the description of the PR to clarify

**Comment by raydouglass** [comment]
rerun tests

**Comment by mike-wendt** [comment]
Doesn't look like we're needed for code review, removing  @rapidsai/ops-codeowners; add us back if necessary. Thanks!

File: cpp/include/raft/cudart_utils.h [hunk 1]
```diff
@@ -256,4 +259,107 @@ void print_device_vector(const char* variable_name, const T* devMem,
...
+  return rows * columns;
+}
+
+/** Helper function to check alignment of pointer.
+    * @param ptr the pointer to check
+    * @param alignment to be checked for
+    * @return true if address in bytes is a multiple of alignment
+    */
+template <typename Type>
+bool is_aligned(Type* ptr, size_t alignment) {
```
Comment (JohnZed) [inline cpp/include/raft/cudart_utils.h:346]: It is pretty fun seeing these varying naming conventions together. Will be great to get through the tidy phase!

**Review by JohnZed — APPROVED** [review]
Looks straightforward to me! Thanks for making the move.

**Comment by divyegala** [comment]
@JohnZed added the leftover prims, as I mentioned on the call. Reference cuml PR is https://github.com/rapidsai/cuml/pull/3044

**Review by afender — APPROVED** [review]

**Comment by JohnZed** [comment]
GIven approvals from both ML and graph, I'll go ahead and merge. Excellent stuff, @divyegala !

**Pull request merged**